generator/src/test/java/com/scottlogic/deg/generator/cucumber/OfLength.feature

```diff
@@ -1,593 +1,455 @@
+       | "1E+02"                   |
+       | "001 000"                 |
+       | "2010-01-01T00:00:00.000" |
+       | "2010-01-01T00:00"        |
+       | "1st Jan 2010"            |
+       | "a"                       |
+       | ""                        |
+
+# COMBINATION OF CONSTRAINTS #
+
+Scenario: ofLength run against a non contradicting ofLength should be successful
+     Given foo is of length 1
+       And foo is of length 1
+       And foo is in set:
+         | "1"  |
+         | "a"  |
+         | "22" |
+     Then the following data should be generated:
+       | foo  |
+       | null |
+       | "1"  |
+       | "a"  |
+
+Scenario: ofLength run against a non contradicting not ofLength should be successful
+     Given foo is of length 1
+       And foo is anything but of length 2
+       And foo is in set:
+         | "1"  |
+         | "a"  |
+         | "22" |
+     Then the following data should be generated:
+       | foo  |
+       | null |
+       | "1"  |
+       | "a"  |
 
-  Scenario: Running an 'ofLength' request that includes a numeric value > 8k on a roman alphabet character string value with length > 8k should be successful
-    Given foo is of length 8001
-    And foo is in set:
-      | "8001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180018001800180
```

Comment:
So this currently works as string constraints (of length e.g.) are never contradictory with numeric constraints (greaterThanOrEqualTo). Do we get much value by confirming this for each numeric constraint and its negation? If we do want to keep these non-contradictory tests for each numeric constraint can they all be in a scenario outline structure instead?

Reply:
This is a regression CI so provides us with confidence that any code changes we've made do not impact the behaviour of each numeric constraint and its negation.